spinoso-time/src/time/tzrs/timezone.rs

```diff
@@ -0,0 +1,155 @@
+use tz::datetime::DateTime;
+
+use super::{Offset, Time};
+
+// Timezone conversions (returns new Time)
+impl Time {
+    /// Returns a new Time object representing _time_ based on the provided offset
+    ///
+    /// Can be used to implement [`Time#getlocal`] with a string/number parameter
+    ///
+    /// #Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let local_offset = Time::now().utc_offset();
+    /// let now_utc = Time::utc(2022, 7, 8, 12, 34, 56, 0);
+    /// let now_local = now_utc.to_local();
+    /// assert_eq!(now_local.utc_offset(), local_offset);
+    /// ```
+    ///
+    /// [`Time#getlocal`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-getlocal
+    #[inline]
+    pub fn to_offset(&self, offset: Offset) -> Self {
+        Self::with_timespec_and_offset(self.inner.unix_time(), self.inner.nanoseconds(), offset)
+    }
+
+    /// Returns a new _time_ in UTC
+    ///
+    /// Can be used to implement [`Time#getutc`] and [`Time#getgm`]
+    ///
+    /// #Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let now_local = Time::now();
+    /// let now_utc = now_local.to_utc();
+    /// assert_eq!(now_utc.utc_offset(), 0);
+    /// ```
+    ///
+    /// [`Time#getutc`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-getutc
+    /// [`Time#getgm`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-getgm
+    #[inline]
+    pub fn to_utc(&self) -> Self {
+        self.to_offset(Offset::utc())
+    }
+
+    /// Returns a new Time object representing _time_ in local time (using the local time zone in
+    /// effect for this process)
+    ///
+    /// Can be used to implement [`Time#getlocal`]
+    ///
+    /// #Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let local_offset = Time::now().utc_offset();
+    /// let now_utc = Time::utc(2022, 7, 8, 12, 34, 56, 0);
+    /// let now_local = now_utc.to_local();
+    /// assert_eq!(now_local.utc_offset(), local_offset);
+    /// ```
```

Comment:
this is a nice example that avoids test time indeterminism

Reply:
Thanks! Write once, run everywhere right? :sweat_smile: